packages/deployments/contracts/contracts/messaging/libraries/Merkle.sol

```diff
@@ -16,6 +16,48 @@ library MerkleLib {
   uint256 internal constant TREE_DEPTH = 32;
   uint256 internal constant MAX_LEAVES = 2**TREE_DEPTH - 1;
 
+  /**
+   * @dev Z_i repsesent the hash values at different heights for a binary tree with leaf values equal to `0`.
+   * (e.g. Z_1 is the keccak256 hash of (0x0, 0x0), Z_2 is the keccak256 hash of (Z_1, Z_1), etc...)
+   * Z_0 is the bottom of the 32-layer tree, Z_31 is the top.
```

Comment:
If we would consider the root layer to be the top layer than we have `33` layers and the top one would be `Z_32`.